src/main/kotlin/org/opensearch/replication/action/replay/TransportReplayChangesAction.kt

```diff
@@ -133,6 +135,9 @@ class TransportReplayChangesAction @Inject constructor(settings: Settings, trans
         val response = ReplayChangesResponse() // TODO: Figure out what to add to response
         return WritePrimaryResult(request, response, location, null, primaryShard, log)
     }
+    fun shouldSyncMappingAndRetry(result: Engine.Result): Boolean {
+        return result.resultType == Engine.Result.Type.MAPPING_UPDATE_REQUIRED || result.failure is MapperParsingException
+    }
```

Comment:
1. Can we add context as code comment?
2. Is the failure already logged? If not, can we ensure to log it?

Reply:
Adding the comment. Failure already gets logged.